components/apimgt/org.wso2.carbon.apimgt.core/src/main/java/org/wso2/carbon/apimgt/core/dao/impl/ApiDAOImpl.java

```diff
@@ -414,14 +431,19 @@ public String getLastUpdatedTimeOfGatewayConfig(String apiId) throws APIMgtDAOEx
             }
 
             statement.setString(++i, ApiType.STANDARD.toString());
+            //Set the label names in the query
+            for (String label : labels) {
+                statement.setString(++i, label);
+            }
 
             return constructAPISummaryList(connection, statement);
         } catch (SQLException e) {
-            throw new APIMgtDAOException(DAOUtil.DAO_ERROR_PREFIX + "getting APIs by Status by Roles", e);
+            String errorMessage = "Error while retrieving API list in store.";
+            log.error(errorMessage, e);
+            throw new APIMgtDAOException(errorMessage, e);
```

Comment:
You have logged the erro msg as well as the exception. Also, you have passed the exception to the next level. From the top level it repeats the same stacktrace and it becomes duplicate. We can do the following to avoid these.

Option 1:, We do not log. But we throw the exception with a msg. 
throw new APIMgtDAOException(errorMessage, e);

Option 2: Log the exception with a msg and throw only the msg.
 log.error(errorMessage, e);
 throw new APIMgtDAOException(errorMessage);

@harsha89 , @sanjeewa-malalgoda , @nuwand : Please advise? Also, we discussed to go with Option 1 as a practice.

Reply:
done